File: src/NTIX.Core/Models/NTIXConfig.cs
namespace NTIX.Core.Models;

public record NTIXConfig(
    NTIXOptions Options,
    List<PackageEntry> WingetPackages = default!,
    List<PackageEntry> ChocoPackages = default!,
    List<PackageEntry> ScoopPackages = default!
)
{
    public List<PackageEntry> WingetPackages { get; init; } = WingetPackages ?? new();
    public List<PackageEntry> ChocoPackages { get; init; } = ChocoPackages ?? new();
    public List<PackageEntry> ScoopPackages { get; init; } = ScoopPackages ?? new();
}

File: src/models/ntix_config.rs
use crate::models::{
    config_file::ConfigFileEntry, import_node::ImportNode, package_entry::PackageEntry,
};

use super::options::NTIXOptions;

#[derive(Debug, Clone, Default)]
pub struct NTIXConfig {
    pub options: NTIXOptions,
    pub winget_packages: Vec<PackageEntry>,
    pub choco_packages: Vec<PackageEntry>,
    pub scoop_packages: Vec<PackageEntry>,
    pub config_files: Vec<ConfigFileEntry>,
    pub imports: Vec<ImportNode>,
}
